runtime/datetime.cpp

```diff
@@ -11,17 +11,29 @@
 
 #include "runtime/critical_section.h"
 #include "runtime/string_functions.h"
+#include "runtime/timelib_wrapper.h"
 
 extern long timezone;
 
+static const char *default_timezone_id = "";
+
+// this function should only be called with supported timezone_id values;
+// it could be "Etc/GMT-3" or "Europe/Moscow"
+static void set_default_timezone_id(const char *timezone_id) {
+  if (strcmp(default_timezone_id, timezone_id) == 0) {
+    return;
+  }
+  setenv("TZ", timezone_id, 1);
```

Comment:
Please, consider using the critical guard section here. I'm not sure, if it is safe to interrupt `setenv()` and `tzset()` calls.